### DISCUSSION TITLE
Missing permissions (HotKey)

### DISCUSSION THREAD

**Discussion by mintgitb1ker** [StreamController/StreamController#213] [Q&A] [answered]
Having a problem with the above.
Followed the new instructions: - https://github.com/streamcontroller/osplugin?tab=readme-ov-file#hotkeys--write-text
Now I copied and pasted exactly what was there and am guessing something was a variable to my specific system - but I don't know which bit
Thx

**Comment by Core447** [comment]
Did you tried the updated instructions from [this](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/231#discussioncomment-10429672) message?

**Reply by mintgitb1ker** [reply]
yes
copied and pasted exactly what was there and rebooted
still same permissions issue

**Reply by Core447** [reply]
Can you please run `groups`? Is `input` in the list?

**Comment by mintgitb1ker** [comment]
Real newbie here. What is that

**Reply by Core447** [reply]
It's a command that prints the groups your user is in. Ideally it should contain `input`.

Just to make sure: I updated the instructions you followed, can you please try the new ones?

**Reply by mintgitb1ker** [reply]
just tried
1st line produces
KERNEL==uinput,: command not found

**Reply by Core447** [reply]
The first line is not a command, but a udev rule. Please follow these steps:
1. Run  `sudo nano /etc/udev/rules.d/99-uinput.rules`
2. Paste `KERNEL=="uinput", SUBSYSTEM=="misc", OPTIONS+="static_node=uinput", TAG+="uaccess", GROUP="input", MODE="0660"` via CTRL + SHIFT + V
3. Save the file by pressing CTRL + S, then Return, then CTRL + X
4. Run `sudo groupadd input` followed by `sudo usermod -aG input $USER`

You can read more about udev rules [here](https://linuxconfig.org/tutorial-on-how-to-write-basic-udev-rules-in-linux).

**Reply by mintgitb1ker** [reply]
thx so much for your help - the strange thing is the hotkeys worked foer a while before that error message.
Still not quite there though:
$ sudo usermod -ag input $USER
usermod: -a flag is only allowed with the -G flag
Usage: usermod [options] LOGIN

**Reply by Core447** [reply, marked as answer]
> thx so much for your help

No problem.

Does it work when using `-aG` instead of `-a`?

**Reply by mintgitb1ker** [reply]
Thx so much - its all working now!!

**Reply by Core447** [reply]
> Thx so much - its all working now!!

Great! Gonna update the command to include the `G`.

**Comment by mintgitb1ker** [comment]
If tu ever need a dumb tester. Let me know!
